packages/app/src/cli/services/context.ts

```diff
@@ -50,6 +50,131 @@ export const InvalidApiKeyErrorMessage = (apiKey: string) => {
   }
 }
 
+export interface LogsContextOptions {
+  directory: string
+  apiKey?: string
+  storeFqdn?: string
+  reset: boolean
+  developerPlatformClient: DeveloperPlatformClient
+}
+
+interface LogsContextOutput {
+  remoteApp: Omit<OrganizationApp, 'apiSecretKeys'> & {apiSecret?: string}
+  storeFqdn: string
+  storeId: string
+  localApp: AppInterface
+}
+
+export async function ensureLogsContext(options: LogsContextOptions): Promise<LogsContextOutput> {
```

Comment:
Was this replicated from what we're doing in dev? 👀

This feels like something that should be behind some kind of shared method so we don't need to implement it all just for logs- but curious to see what CLI folks think.

Reply:
Yea, mostly just copied over the same code. I can look into this more.

Reply:
Yes, can you ideally just use the `ensureDevContext` function directly? Its job is to take the command flags and produce all the options the command itself will need to run (e.g. loading the app). We definitely don't want to duplicate here

Reply:
To expand on this though. The `logs` command is similar to `dev` in that its showing this growing output of activity on a shop. But on the other hand it's extremely different: we expect developer to use `dev` all the time and that it would be the command that might do things like guide a developer through selecting a shop/app record (that's a lot of what happens here). Is that the case for `logs`? It seems like you need very little: an api key and the shop's ID. So maybe the right thing is something substantially smaller.